src/components/asthma/components/AsthmaBiometrics/AsthmaBiometrics.tsx

```diff
@@ -0,0 +1,135 @@
+    let [nighttimeRestingHeartRate, setNighttimeRestingHeartRate] = useState<AsthmaBiometric>();
+    let [respiratoryRate, setRespiratoryRate] = useState<AsthmaBiometric>();
+    let [steps, setSteps] = useState<AsthmaBiometric>();
+    let [sleepDisturbances, setSleepDisturbances] = useState<AsthmaBiometric>();
+    let [daytimeBloodOxygenLevel, setDaytimeBloodOxygenLevel] = useState<AsthmaBiometric>();
+    let [nighttimeBloodOxygenLevel, setNighttimeBloodOxygenLevel] = useState<AsthmaBiometric>();
+
+    useInitializeView(() => {
+        setLoading(true);
+
+        if (props.previewState === 'loading') {
+            return;
+        }
+        if (props.previewState) {
+            setDaytimeRestingHeartRate(previewData[props.previewState].daytimeRestingHeartRate);
+            setNighttimeRestingHeartRate(previewData[props.previewState].nighttimeRestingHeartRate);
+            setRespiratoryRate(previewData[props.previewState].respiratoryRate);
+            setSteps(previewData[props.previewState].steps);
+            setSleepDisturbances(previewData[props.previewState].sleepDisturbances);
+            setDaytimeBloodOxygenLevel(previewData[props.previewState].daytimeBloodOxygenLevel);
+            setNighttimeBloodOxygenLevel(previewData[props.previewState].nighttimeBloodOxygenLevel);
+            setLoading(false);
+            return;
+        }
+
+        let biometricsQuery = props.date ? asthmaDataService.loadBiometricsForDate(props.date) : asthmaDataService.loadBiometricsForControlStatus();
+
+        biometricsQuery.then(biometrics => {
+            setDaytimeRestingHeartRate(biometrics.find(b => b.type === 'daytime-resting-heart-rate'));
+            setNighttimeRestingHeartRate(biometrics.find(b => b.type === 'nighttime-resting-heart-rate'));
+            setRespiratoryRate(biometrics.find(b => b.type === 'respiratory-rate'));
+            setSteps(biometrics.find(b => b.type === 'steps'));
+            setSleepDisturbances(biometrics.find(b => b.type === 'sleep-disturbances'));
+            setDaytimeBloodOxygenLevel(biometrics.find(b => b.type === 'daytime-blood-oxygen-level'));
+            setNighttimeBloodOxygenLevel(biometrics.find(b => b.type === 'nighttime-blood-oxygen-level'));
+            setLoading(false);
+        });
+    }, [], [props.previewState]);
+
+    const onClick = (type: AsthmaBiometricType): void => {
+        if (props.previewState) return;
+        if (type === 'daytime-resting-heart-rate' || type === 'nighttime-resting-heart-rate' || type === 'respiratory-rate' || type === 'daytime-blood-oxygen-level' || type === 'nighttime-blood-oxygen-level') {
+            MyDataHelps.openApplication(props.heartAndLungsUrl, {modal: true});
+        } else if (type === 'steps') {
+            MyDataHelps.openApplication(props.activityUrl, {modal: true});
+        } else if (type === 'sleep-disturbances') {
+            MyDataHelps.openApplication(props.sleepUrl, {modal: true});
```

Comment:
Yeah here's  another place you could have these be event handlers that go to an ActionDefinition in the view builder.  Don't feel super strongly about it though

Reply:
As we discussed, I'm going to punt on the inversion of control (event handler/action definition) stuff for now until I can think through the best way to apply it across the components with some level of consistency.

At present, the components are less flexible in that they don't allow for that level of customization, but they are at least consistent in being that opinionated about how each action should be handled.  I don't think that all of the current places where we'd make that change will be able to be converted, so it would have us in a state where some events are handled internal and some externally (even within the same component), which feels less warm and fuzzy to me.  😃